tests/integration.rs

```diff
@@ -4356,6 +4372,68 @@ mod tests {
             handle_child_output(r, &output);
         }
 
+        #[test]
+        #[cfg(target_arch = "x86_64")]
+        fn test_virtio_balloon() {
+            let focal = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_string());
+            let guest = Guest::new(Box::new(focal));
+
+            let kernel_path = direct_kernel_boot_path();
+
+            let api_socket = temp_api_path(&guest.tmp_dir);
+
+            let guest_memory_size_gb = 4;
+
+            //Let's start a 4G guest with balloon occupied 2G memory
+            let mut child = GuestCommand::new(&guest)
+                .args(&["--api-socket", &api_socket])
+                .args(&["--cpus", "boot=4"])
+                .args(&[
+                    "--memory",
+                    format!("size={}G", guest_memory_size_gb).as_str(),
+                ])
+                .args(&["--kernel", kernel_path.to_str().unwrap()])
+                .args(&["--cmdline", DIRECT_KERNEL_BOOT_CMDLINE])
+                .args(&["--balloon", "size=2147483648,deflate_on_oom=on"])
+                .default_disks()
+                .default_net()
+                .capture_output()
+                .spawn()
+                .unwrap();
+
+            let r = std::panic::catch_unwind(|| {
+                guest.wait_vm_boot(None).unwrap();
+
+                // Wait for balloon memory's initialization, and then check its size
+                thread::sleep(std::time::Duration::new(5, 0));
+
+                let orig_balloon = get_balloon_mem(&api_socket);
+                println!("The original balloon memory size is {} bytes", orig_balloon);
+                assert!(orig_balloon == 2147483648);
+
+                // Two steps to verify if the 'deflate_on_oom' parameter works.
+                // 1st: run a command in guest to eat up memory heavily, which
+                // will consume much more memory than $(total_mem - balloon_mem)
+                // to trigger an oom.
+                guest
+                    .ssh_command("sudo stress --vm 25 --vm-keep --vm-bytes 1G --timeout 20")
+                    .unwrap();
+
+                // 2nd: check balloon_mem's value to verify balloon has been automatically deflated
+                let deflated_balloon = get_balloon_mem(&api_socket);
+                println!(
+                    "After deflating, balloon memory size is {} bytes",
+                    deflated_balloon
+                );
+                assert!(deflated_balloon > 0 && deflated_balloon < 2147483648);
```

Comment:
Let's check this a bit more by making sure the balloon deflated by 10% at least, leading to the following:
```suggestion
                // Verify the balloon size deflated by 10% at least
                assert!(deflated_balloon > 0 && deflated_balloon < 1932735283);
```